pkg/kinflate/transformers/mergekobject.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformers
+
+import (
+	"k8s.io/kubectl/pkg/kinflate/types"
+)
+
+// MergeKObjectTransformer contains a KObject that will be merged with another KObject.
+type MergeKObjectTransformer struct {
+	kobj types.KObject
+}
+
+var _ Transformer = &MergeKObjectTransformer{}
+
+// NewMergeKObjectTransformer constructs a MergeKObjectTransformer.
+func NewMergeKObjectTransformer(kobj types.KObject) (Transformer, error) {
+	if len(kobj) == 0 {
+		return NewNoOpTransformer(), nil
+	}
+	return &MergeKObjectTransformer{kobj: kobj}, nil
+}
+
+// Transform merges 2 KObjects
+func (o *MergeKObjectTransformer) Transform(m types.KObject) error {
+	return types.Merge(m, o.kobj)
+}
```

Comment:
Not very clear about the motivation....why would this be a transformer ?

Reply:
Every single manipulation on the KObject is now a transformer.
No need to do preprocessing:
```
Merge(Resources, ConfigMaps)
Merge(Resources, Secrets)
Transform(Resources)
```